x-pack/plugins/security/server/deprecations/kibana_dashboard_only_role.ts

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  SecurityGetRoleMappingResponse,
+  SecurityGetUserResponse,
+} from '@elastic/elasticsearch/api/types';
+
+import { i18n } from '@kbn/i18n';
+import type {
+  DeprecationsDetails,
+  DeprecationsServiceSetup,
+  ElasticsearchClient,
+  Logger,
+  PackageInfo,
+} from 'src/core/server';
+
+import type { SecurityLicense } from '../../common';
+import { getDetailedErrorMessage, getErrorStatusCode } from '../errors';
+
+export const KIBANA_DASHBOARD_ONLY_USER_ROLE_NAME = 'kibana_dashboard_only_user';
+
+export interface Deps {
+  deprecationsService: DeprecationsServiceSetup;
+  license: SecurityLicense;
+  logger: Logger;
+  packageInfo: PackageInfo;
+}
+
+export const registerKibanaDashboardOnlyRoleDeprecation = ({
+  deprecationsService,
+  logger,
+  license,
+  packageInfo,
+}: Deps) => {
+  deprecationsService.registerDeprecations({
+    getDeprecations: async (context) => {
+      // Nothing to do if security is disabled
+      if (!license.isEnabled()) {
+        return [];
+      }
+
+      return [
```

Comment:
__note:__ I split role mappings and users related cases in two separate deprecation entries. I don't like that it has the same name in the UA table, but apart from that the finer granularity fits nicely here (since user needs to use different screens during manual resolution).